compiler/src/iree/compiler/Codegen/LLVMGPU/test/vector_lowering.mlir

```diff
@@ -111,9 +111,8 @@ func.func @multi_reduction_f32(%a: vector<2x1x8xf32>, %b: vector<2x1x8xf32>) ->
 // CHECK-LABEL: func.func @multi_reduction_f32
 // CHECK-DAG:  %[[C0:.+]]  = arith.constant 0.000000e+00 : f32
 // CHECK-DAG:  %[[V0:.+]]  = arith.constant dense<0.000000e+00> : vector<2x1x8xf32>
-// CHECK:      %[[MUL:.+]] = arith.mulf %{{.*}}, %{{.*}} : vector<2x1x8xf32>
-// CHECK:      %[[ADD:.+]] = arith.addf %[[MUL]], %[[V0]] : vector<2x1x8xf32>
-// CHECK:      %[[E0:.+]]  = vector.extract %[[ADD]][0, 0] : vector<8xf32> from vector<2x1x8xf32>
+// CHECK:      %[[FMA:.+]] = math.fma %{{.*}}, %{{.*}}, %[[V0]] fastmath<contract> : vector<2x1x8xf32>
+// CHECK:      %[[E0:.+]]  = vector.extract %[[FMA]][0, 0] : vector<8xf32> from vector<2x1x8xf32>
 // CHECK:                    vector.reduction <add>, %[[E0]], %[[C0]] : vector<8xf32> into f32
-// CHECK:      %[[E1:.+]]  = vector.extract %[[ADD]][1, 0] : vector<8xf32> from vector<2x1x8xf32>
+// CHECK:      %[[E1:.+]]  = vector.extract %[[FMA]][1, 0] : vector<8xf32> from vector<2x1x8xf32>
```

Comment:
Can you add a testcase where there are some existing fast math flags and we don't uplift to fma?